src/events/events.ts

```diff
@@ -25,13 +25,21 @@ class CustomEvent<DataType> extends Event {
 }
 
 export abstract class BaseEvent<DataType> extends CustomEvent<DataType> {
-  constructor(data?: DataType) {
+  constructor(data: DataType) {
     super(EventType, { detail: data });
   }
 }
 
-export class ComputationStarted extends BaseEvent<undefined> {}
-export class ComputationFinished extends BaseEvent<undefined> {}
+export class ComputationStarted extends BaseEvent<undefined> {
+  constructor() {
```

Comment:
Short answer: To avoid having to do `new Events.ComputationStarted(undefined)`

Long answer:  In order to satisfy `detail` always being of type `DataType` and not `DataType | undefined`:
```ts
class CustomEvent<DataType> extends Event {
  readonly detail: DataType; // <--------------
  // ... rest
}
```
we need the `BaseEvent`'s constructor to always take an argument
```ts
export abstract class BaseEvent<DataType> extends CustomEvent<DataType> {
  constructor(data: DataType) { // <------ this cannot optional because `undefined` may not be a valid `DataType`
    super(EventType, { detail: data });
  }
}
```
which is what we want

but if you set the `DataType = undefiend` like in the case of `ComputationStarted` then you'd be forced to pass a parameter of that type to the constructor every time you create this event, resulting in having to do `new Events.ComputationStarted(undefined)` which is ugly and confusing. The easy fix for this edge case is just to overwrite the constructor to not take any parameters for this particular event